### DISCUSSION TITLE
Question on wayland

### DISCUSSION THREAD

**Discussion by Zerogaku** [input-leap/input-leap#1730] [Q&A] [answered]
### What happened?

I would like to know if wayland support exists now, from what i've seen people can/or attempt to use input leap on wayland devices but i haven't found anything saying if it already has support, only the long open issue thread talking about adding wayland.

### Version

v2.0.0-RC1

### If applicable, where did you install Input Leap from?

Package manager

### What OSes are you seeing the problem on? (Check all that apply)

Linux

### If applicable, are you using Wayland or X11?

X11 but also have a wayland installation

### What OS versions are you using?

Artix linux

**Comment by gabeklavans** [comment]
You can try building and testing yourself :P but as far as I can tell, no, there is still no fully-fledged wayland support here or on any virtual KVM software. Some workaround exist to get clients working, I believe.

**Reply by shymega** [reply, marked as answer]
That's incorrect I'm afraid. We do have fully fledged Wayland support via `libei`. However, it's dependent on each _individual compositor/DE/WM_ to implement support on their end.

GNOME for example, now supports `libei`. But it's early days.

Also, I would just like to add to OP: please, read the issue template. It says twice there to not open questions as an issue - and instead use the Discussions tab. I had to convert your issue manually, which means more work for maintainers.

**Reply by gabeklavans** [reply]
Wow yeah, didn't realize all the activity that happened for Wayland since I last checked. I've got some reading to do!
